docs(telemetry): rewrite for Scarf + embed ecosystem pixel in README (PACT-7009) (#419)
pact-foundation/docs.pact.io: website/docs/telemetry.md

The Pact team strive to develop great open source tools for the community. We rely on the community's input to help us iterate on and improve Pact. Telemetry is additional information that helps us to better understand the community's needs, diagnose issues, and prioritise feature work.

Pact's distribution channels — Docker images on Docker Hub, CLI binaries on GitHub Releases, and similar — are routed through [Scarf](https://scarf.sh/), which collects anonymized usage analytics about how Pact tools are downloaded. README impressions on github.com and Docker Hub are also tracked via 1x1 pixel images served from `static.scarf.sh`.

We do not collect personal information, such as usernames or email addresses. Pact tools do not extract sensitive project-level information at runtime.

When consuming Pact tools you can control whether telemetry is enabled, and the setting can be changed at any point in time. If telemetry remains enabled, telemetry data is sent in the background without requiring any additional interaction.

> This data is collected via [Scarf](https://scarf.sh/) and is accessible by Pact OSS maintainers.

Reviewer: ```suggestion
This data is collected via [Scarf](https://scarf.sh/) and is accessible to SmartBear.
```

## Opt Out

> Scarf collects analytics at the moment you pull / download a Pact distribution. The primary opt-out is **to use the canonical (Docker Hub / GitHub Releases / etc.) URLs directly instead of the Pact Foundation's Scarf gateway URLs documented in each project's README.**

Reviewer: ```suggestion
Scarf collects analytics at the moment you install or download a Pact distribution. The primary opt-out is **to use the canonical (Docker Hub / GitHub Releases / etc.) URLs directly instead of the Pact Foundation's Scarf gateway URLs documented in each project's README.**
```
